avocado/utils/network.py

```diff
@@ -71,3 +73,44 @@ def find_free_ports(start_port, end_port, count, address="localhost"):
             count -= 1
         i += 1
     return ports
+
+
+class PortTracker(Borg):
+
+    """
+    Tracks ports used in the host machine.
+    """
+
+    def __init__(self):
+        Borg.__init__(self)
+        self.address = 'localhost'
```

Comment:
Well this seems a bit limiting, one might be using specific address/interface and/or keep track of multiple different hosts as well (without the feature to get free ports or with a different implementation).